### ISSUE TITLE
RC3: graphchi small transiently fails validation

### ISSUE THREAD

**Issue by farquet** [dacapobench/dacapobench#233]
With different VMs and on different hardware (big linux servers or Mac laptop), I could reproduce the following validation error that happens randomly throughout `graphchi` execution. It only happens for the `small` workload size though and can be reproduced once in a few runs.
```
===== DaCapo 23.9-RC3-chopin graphchi starting warmup 10 =====
Found shards -- no need to preprocess
Set latent factor dimension to: 20
Initializing latent factors for 9798 vertices
Creating 1 blocks
Total edges: 10000
195940 elements, check: 3
Digest validation failed for stdout.log, expecting 0xce90a8af938806c984cdafa69732ecd6045aa4d3 found 0xd32c10c5ac26cfabac154fe35d2224c7dcda3b1c
===== DaCapo 23.9-RC3-chopin graphchi FAILED warmup =====
===== DaCapo 23.9-RC3-chopin graphchi starting warmup 11 =====
Found shards -- no need to preprocess
Set latent factor dimension to: 20
Initializing latent factors for 9798 vertices
Creating 1 blocks
Total edges: 10000
195940 elements, check: 2
===== DaCapo 23.9-RC3-chopin graphchi completed warmup 11 in 422 msec =====
```

Trying with `-v --preserve` couldn't help debug further since it seems that only the last iteration output is preserved.
The validation report doesn't give more info than the incorrect digest already visible in the console.

**Comment by steveblackburn** [comment]
Thanks.  I will investigate further.

**Comment by steveblackburn** [comment]
The checksum was not robust.  Fixed in ae40d2dcbb5ddbae0f96d03496b80fab6a2fa4ee.
